qcengine/programs/openmm.py

```diff
@@ -0,0 +1,239 @@
+        parent = config.scratch_directory
+        with temporary_directory(parent=parent, suffix="_openmm_scratch") as tmpdir:
+
+            # Grab molecule, forcefield
+            jmol = input_data.molecule
+            
+            # TODO: If urls are supported by
+            # `openforcefield.typing.engines.smirnoff.ForceField` already, we
+            # can eliminate the `offxml` and `url` distinction
+            # URL processing can happen there instead
+            if getattr(input_data.model, 'offxml', None):
+                # we were given a file path or relative path
+                offxml = input_data.model.offxml
+
+                # Load an Open Force Field `ForceField`
+                off_forcefield = self._get_off_forcefield(offxml, offxml)
+            elif getattr(input_data.model, 'url', None):
+                # we were given a url
+                with urllib.request.urlopen(input_data.model.url) as req:
+                    xml = req.read().decode()
+
+                # Load an Open Force Field `ForceField`
+                with io.StringIO(xml) as f:
+                    off_forcefield = self._get_off_forcefield(xml, f)
+            else:
+                raise InputError("OpenMM requires either `model.offxml` or `model.url` to be set")
+
+            # Process molecule with RDKit
+            rdkit_mol = RDKitHarness._process_molecule_rdkit(jmol)
+
+            # Create an Open Force Field `Molecule` from the RDKit Molecule
+            off_mol = offtop.Molecule(rdkit_mol)
+
+            # Create OpenMM system in vacuum from forcefield, molecule
+            off_top = off_mol.to_topology()
+            openmm_system = self._get_openmm_system(off_forcefield, off_top)
+            
+            # Need an integrator for simulation even if we don't end up using it really
+            integrator = openmm.VerletIntegrator(1.0*unit.femtoseconds)
+
+            # Set platform to CPU explicitly
+            platform = openmm.Platform.getPlatformByName('CPU')
+
+            # Initialize context
+            context = openmm.Context(openmm_system, integrator, platform)
+
+            # TODO: FIXME set number of threads;
+            # this line currently throws an exception
+            if nthreads:
+                platform.setPropertyValue(context, 'Threads', str(nthreads))
+
+            # Set positions from our Open Force Field `Molecule`
+            context.setPositions(off_mol.conformers[0])
+
+            ret_data["properties"] = {}
+
+            # Execute driver
+            try:
+                if input_data.driver == "energy":
+
+                    # Compute the energy of the configuration
+                    state = context.getState(getEnergy=True)
+
+                    # Get the potential as a simtk.unit.Quantity, put into units of hartree
+                    q = (state.getPotentialEnergy() / unit.hartree)
+                    ret_data["return_result"] = q.value_in_unit(q.unit)
```

Comment:
We always need the energy, should go into `properties.return_energy`.

Reply:
Done!